rcl/src/rcl/timer.c

```diff
@@ -50,26 +52,29 @@ rcl_get_zero_initialized_timer()
 rcl_ret_t
 rcl_timer_init(
   rcl_timer_t * timer,
+  rcl_clock_t * clock,
   int64_t period,
   const rcl_timer_callback_t callback,
   rcl_allocator_t allocator)
 {
   RCL_CHECK_ALLOCATOR_WITH_MSG(&allocator, "invalid allocator", return RCL_RET_INVALID_ARGUMENT);
   RCL_CHECK_ARGUMENT_FOR_NULL(timer, RCL_RET_INVALID_ARGUMENT, allocator);
+  RCL_CHECK_ARGUMENT_FOR_NULL(clock, RCL_RET_INVALID_ARGUMENT, allocator);
   RCUTILS_LOG_DEBUG_NAMED(ROS_PACKAGE_NAME, "Initializing timer with period: %" PRIu64 "ns", period)
   if (timer->impl) {
     RCL_SET_ERROR_MSG("timer already initailized, or memory was uninitialized", allocator);
     return RCL_RET_ALREADY_INIT;
   }
-  rcl_time_point_value_t now_steady;
-  rcl_ret_t now_ret = rcutils_steady_time_now(&now_steady);
+  rcl_time_point_value_t now;
+  rcl_ret_t now_ret = clock->get_now(clock->data, &now);
```

Comment:
I think `rcl_clock_get_now` is the expected interface for getting a clock's time value

Reply:
The commit e8f2e34b15951c86658a2ef3a0756059e3627f93 uses your proposed function.

It requires to get a `rcl_time_point_t` instead of a `rcl_time_point_value_t`. Since the code doesn't care / check the clock type of that time point wouldn't it be more efficient to just get the `rcl_time_point_value_t`? Not sure if that is a good idea...

Reply:
I raised the concern about the existing API in #273 in order to keep it separate from this patch.

Reply:
I rolled the change related to #273 into this patch since both affect the same function signature: 28e6ba1d96fb78798d93719c02b78d49d797f36c